### DISCUSSION TITLE
WebhookSettings and Id

### DISCUSSION THREAD

**Discussion by saan800** [Jericho/StrongGrid#511] [Q&A] [answered]
Hi Jericho

First off, may I say I am so happy i found your library, it looks like its going to save me a lot of time.

I had a couple of questions on how the WebhookSettings works for managing multiple webhooks.

Ref:
* [Code: WebhookSettings.cs](https://github.com/Jericho/StrongGrid/blob/main/Source/StrongGrid/Resources/WebhookSettings.cs)
* [Code: EventWebhookSettings.cs](https://github.com/Jericho/StrongGrid/blob/main/Source/StrongGrid/Models/EventWebhookSettings.cs)
* [Doc: Get All Event Webhooks](https://docs.sendgrid.com/api-reference/webhooks/get-all-event-webhooks)
* [Doc: Get An Event Webhook](https://docs.sendgrid.com/api-reference/webhooks/get-an-event-webhook)
* [Doc: Update an Event Webhook](https://docs.sendgrid.com/api-reference/webhooks/update-an-event-webhook)

In the get and update functions in WebhookSettings or EventWebhookSettings, I can't see any references to an `Id` property.

How do these work when you have more than one webhook setup?
I wouldn't want to get or update everything - just the one object i want

Thanks for your help
Saan

**Comment by Jericho** [comment, marked as answer]
The functionality to update webhook settings was added to StrongGrid back in March 2017. At the time (if I remember correctly) , you couldn't have multiple webhooks configured and there was no "id" to uniquely identify a particular set of webhook settings. This explains why you can't find an Id property on the webhook settings class in the StrongGrid library. In the intervening 7 years, SendGrid has changed their API to allow multiple webhook settings but, as you found out, StrongGrid has not been updated to match this new functionality.

As far as your question about "How do these work when you have more than one webhook setup?", the answer is in SendGrid's documentation:

> If you do not pass a webhook ID to this endpoint, it will update and return your oldest webhook by created_date. This means the default webhook updated by this endpoint when no ID is provided will be the first one you created. This functionality allows customers who do not have multiple webhooks to use this endpoint to update their only webhook, even if they do not supply an ID.

**Reply by saan800** [reply]
Ahh ok, that makes sense. If I get time, I might do a pr to your repo to support both options

Thanks for the quick response

**Reply by Jericho** [reply]
FYI I raised a few issues to track related new functionality/fixes: #512, #513, #514, #516, #517, #518, #519

**Comment by Jericho** [comment]
@saan800 I uploaded a beta package called StrongGrid.0.107.0-multiple-webhook0023 to [my personal NuGet feed](https://feedz.io/org/jericho/repository/jericho) to allow you to manage webhook settings by ID. Let me know if you have a chance to test it and if you have any feedback.

**Reply by Jericho** [reply]
Update: I wrote [brief instructions](https://github.com/Jericho/Jericho/blob/main/MyNugetFeed.md) on how to add my personal feed to your git environment, hope it's helpful.

**Reply by Jericho** [reply]
Version 107 was released some time ago. It includes the enhancements discussed.
